step-functions/nino_issue_credential.asl.json

```diff
@@ -0,0 +1,289 @@
+        "Name": "${JwtTtlUnitParameter}"
+      },
+      "Resource": "arn:aws:states:::aws-sdk:ssm:getParameter",
+      "ResultPath": "$.jwtTTLUnit"
+    },
+    "Fetch VC Issuer": {
+      "Type": "Task",
+      "Parameters": {
+        "Name": "/common-cri-api/verifiable-credential/issuer"
+      },
+      "Resource": "arn:aws:states:::aws-sdk:ssm:getParameter",
+      "Next": "Fetch VC KMS Signing Key",
+      "ResultPath": "$.vc_issuer"
+    },
+    "Fetch VC KMS Signing Key": {
+      "Type": "Task",
+      "Parameters": {
+        "Name": "/common-cri-api/verifiableCredentialKmsSigningKeyId"
+      },
+      "Resource": "arn:aws:states:::aws-sdk:ssm:getParameter",
+      "Next": "Fetch Person Identity Table Name",
+      "ResultPath": "$.kmsSigningKeyId"
+    },
+    "Fetch Person Identity Table Name": {
+      "Type": "Task",
+      "Next": "Fetch Session Table Name",
+      "Parameters": {
+        "Name": "/common-cri-api/PersonIdentityTableName"
+      },
+      "Resource": "arn:aws:states:::aws-sdk:ssm:getParameter",
+      "ResultPath": "$.personIdentityTableName"
+    },
+    "Fetch Session Table Name": {
+      "Type": "Task",
+      "Next": "Query Session Item",
+      "Parameters": {
+        "Name": "/common-cri-api/SessionTableName"
+      },
+      "Resource": "arn:aws:states:::aws-sdk:ssm:getParameter",
+      "ResultPath": "$.sessionTable"
+    },
+    "Query Session Item": {
+      "Type": "Task",
+      "Parameters": {
+        "TableName.$": "$.sessionTable.Parameter.Value",
+        "IndexName": "access-token-index",
+        "KeyConditionExpression": "accessToken = :value",
+        "ExpressionAttributeValues": {
+          ":value": {
+            "S.$": "$.bearerToken"
+          }
+        }
+      },
+      "Resource": "arn:aws:states:::aws-sdk:dynamodb:query",
+      "Next": "Fetch details from person identity",
+      "ResultPath": "$.querySession"
+    },
+    "Fetch details from person identity": {
+      "Type": "Task",
+      "Next": "Create Name Parts",
+      "Parameters": {
+        "TableName.$": "$.personIdentityTableName.Parameter.Value",
+        "KeyConditionExpression": "sessionId = :value",
+        "ExpressionAttributeValues": {
+          ":value": {
+            "S.$": "$.querySession.Items[0].sessionId.S"
+          }
+        }
+      },
+      "Resource": "arn:aws:states:::aws-sdk:dynamodb:query",
+      "ResultPath": "$.userDetails"
+    },
+    "Create Name Parts": {
+      "Type": "Map",
+      "ItemProcessor": {
+        "ProcessorConfig": {
+          "Mode": "INLINE"
+        },
+        "StartAt": "ExtractData",
+        "States": {
+          "ExtractData": {
+            "Type": "Pass",
+            "End": true,
+            "Parameters": {
+              "value.$": "$.M.value.S",
+              "type.$": "$.M.type.S"
+            }
+          }
+        }
+      },
+      "Next": "Create VC Structure",
+      "ItemsPath": "$.userDetails.Items[0].names.L[0].M.nameParts.L",
```

Comment:
This is gross but I'm not sure how else we'd do it 😞 . Is it worth having a choice before this to check we got some user details back, or it'll fail on the `Items[0]` bit?

Reply:
It'll fail on the `Items[0]` bit at the moment - we should cover error handling properly once we're ready to do so.